client/src/modules/stock/entry/modals/lots.modal.js

```diff
@@ -124,6 +126,19 @@ function StockDefineLotsModalController(Instance, Notify, uiGridConstants, Data,
     var hasExpiration = (new Date(inventory.expiration_date) >= new Date());
     inventory.is_valid = (hasQuantity && hasLotLabel && hasExpiration);
 
+    // set the value if the date is not correct
+    vm.isExpiration = hasExpiration ? false : true;
```

Comment:
Hmm... this binds a lot of variables to `vm` that you don't actually need.

How about your write a validation function for an inventory item (or lot, whichever makes sense)?  Something like this:
```js
function hasInvalidInventory(inventory) {
   // returns a string error like 'FORM.ERRORS.PLEASE_CHECK_EXPIRY_DATE' if there is an error.
  // otherwise, returns false.
}
```
You could then just call this function on every inventory item and it would return the appropriate message.
```js
var error = hasInvalidInventory(inventory);
if (error) {
  vm.submitErrror = true;
  vm.errorText = error;
}
```

This would reduce the logic needed to know the error type.  What do you think?